src/panel/direction/road_map_panel.js

```diff
@@ -1,136 +1,35 @@
-import Panel from '../../libs/panel';
-import roadMapTemplate from '../../views/direction/road_map.dot';
-import Device from '../../libs/device';
-import RoadMapPreviewPanel from './road_map_preview';
-import Telemetry from '../../libs/telemetry';
-import { openShareModal } from 'src/modals/ShareModal';
-import { formatDuration, formatDistance } from 'src/libs/route_utils';
+import React from 'react';
+import ReactDOM from 'react-dom';
+import RouteResult from './RouteResult';
 
 export default class RoadMapPanel {
-  constructor(onOpen, onClose) {
-    this.onOpen = onOpen;
-    this.onClose = onClose;
-    this.previewRoadMap = new RoadMapPreviewPanel(this.distance);
-    this.showRoute = true;
-    this.panel = new Panel(this, roadMapTemplate);
-    this.routes = [];
-    this.isMobile = Device.isMobile;
-    this.placeholder = false;
-    this.error = false;
-    this.origin = null;
-
-    listen('select_road_map', i => {
-      this.toggleRoute(i);
-    });
+  constructor(openMobilePreview) {
+    this.onOpenMobilePreview = openMobilePreview;
   }
 
+  getContainer = () => document.getElementById('react_itinerary_result');
+
   setRoad(routes, vehicle, origin) {
-    if (origin) {
-      this.origin = origin;
-    }
-    this.routes = routes.map(roadStep => {
-      return roadStep;
-    });
-    this.vehicle = vehicle;
-    this.error = false;
-    this.panel.update();
+    ReactDOM.render(
+      <RouteResult
+        routes={routes}
+        vehicle={vehicle}
+        origin={origin.getInputValue()}
+        onOpenMobilePreview={this.onOpenMobilePreview}
+      />,
+      this.getContainer()
+    );
   }
 
   showPlaceholder(vehicle) {
-    this.placeholder = true;
-    this.vehicle = vehicle;
-    this.panel.update();
-  }
-
-  hidePlaceholder() {
-    this.placeholder = false;
-    this.panel.update();
+    ReactDOM.render(<RouteResult isLoading vehicle={vehicle} />, this.getContainer());
   }
 
   showError() {
-    this.error = true;
-    this.panel.update();
-  }
-
-  preview() {
-    this.showRoute = false;
-    this.previewRoadMap.setRoad(this.routes.find(route => route.isActive));
-    this.onOpen();
-    this.panel.update();
-    fire('show_marker_steps');
-    document.querySelector('.map_bottom_button_group').classList.add('itinerary_preview--active');
```

Comment:
this class `itinerary_preview--active` was used (in a ugly way...) to move the geolocation button above the previous/next button during preview.

It seems this behavior is lost now: the geolocation button is not visible
![image](https://user-images.githubusercontent.com/4726554/64866893-9bb19880-d63c-11e9-8aa1-f3cd80bd2312.png)

We should also check how it behaves once changes from #365 are applied